SPARKCONF.md

```diff
@@ -0,0 +1,155 @@
+ 
+ >Having a total of 432 vcpus, 48 per node: 432/43 ~ 10 cores per executor. 
+ 
+Although it sounds like a good idea to allocate all the available cores, we have seen cases where many cores 
+per executor will cause Spark to assign more task to every executor and that can potentially cause **OOM** errors. 
+Is recommended to keep a close relation between cores and executor memory. 
+ 
+Lastly, for overhead memory we recommend to use something between 8% and 10% of executor memory.
+ 
+Following the example, the values for the _Spark_ variables in duxbay.conf would look like this:
+ 
+            SPK_EXEC='43'
+            SPK_EXEC_MEM='30475m'
+            SPK_EXEC_CORES='6'
+            SPAK_EXEC_MEM_OVERHEAD='3047'
+ 
+#####Driver Memory, Driver Maximum Results and Driver Memory Overhead
+ 
+ oni-ml application executes actions such as _.collect_, _.orderBy_, _.saveAsTextFile_ so we recommend to assign a 
+ considerable amount of memory for the driver. 
+ 
+ The same way, driver maximum results should be enough for the serialized results.
+ 
+ Depending on users data volume these two properties can be small as tens of gigabytes for driver and a couple of gigabytes for 
+ driver maximum results or grow up to 50 GB and 8 GB respectively. Users can follow the next steps to determine the amount of memory
+ for driver and maximum results:
+ 
+1. If executor memory is equal or bigger than 30 GB, make driver memory the same as executor memory and driver maximum results
+a value equal or bigger than 6 GB.
+2. If executor memory is less than 30 GB but data to be analyzed is equal or bigger than 100 GB make driver something between 30 GB
+and 50 GB. Driver maximum results should be something equal or bigger than 8 GB. 
+ 
+ >Following the example in the previous section, with 9 nodes, 43 executors, 30 GB memory each executor, we can set driver memory
+ to 30GB and 8GB driver maximum results.
+ 
+ Memory overhead for driver can be set to something between 8% and 10% of driver memory.
+ 
+ This is how _Spark_ variables look like for driver properties:
+ 
+        SPK_DRIVER_MEM='30475m'
+        SPK_DRIVER_MAX_RESULTS='8g'
+        SPK_DRIVER_MEM_OVERHEAD='3047'
+        
+ ![DriverMemory](https://raw.githubusercontent.com/Open-Network-Insight/oni-docs/master/images/Driver%20Memory.png)
+ 
+ _Representation of memory allocation in driver node._
+ 
+ 
+ For more information about Spark properties click [here](http://spark.apache.org/docs/latest/configuration.html).
+ 
+###Known Spark error messages running oni-ml
+ 
+####Out Of Memory Error
+ 
+This issue includes _java.lang.OutOfMemoryError: Java heap space_ and _java.lang.OutOfMemoryError : GC overhead limit exceeded_.
+When users get OOME can be for many different issues but we have identified a couple of important points or reasons for this
+error in oni-ml. 
+
+The main reason for this error in oni-ml can be when oni-lda-c (ML part of oni-ml) returns many results for _word probabilities per topic_.
```

Comment:
not sure why oni-lda-c needs to be mentioned... at issue is the size of the broadcasted table, not how it is calculated